File: Data Structures & Algorithms/duplicate-integer/submission-2.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        // Bruteforce - O(n*n) & O(1)
        int n = nums.size();
        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j < n; j++) {
                if (nums[i] == nums[j]) {
                    return true;
                }
            }
        }
        return false;
    }
};

File: Data Structures & Algorithms/first-missing-positive/submission-2.py
class Solution:
    def firstMissingPositive(self, nums: List[int]) -> int:
        numSet = set(nums)
        ansRange = len(nums) + 1
        # Mathematically, ans will belong to [1, len(nums) + 1]
        for i in range(1, ansRange):
            if i not in numSet:
                return i

        return ansRange
        